src/libcrun/linux.c

```diff
@@ -5617,6 +5617,8 @@ libcrun_kill_linux (libcrun_container_status_t *status, int signal, libcrun_erro
       /* If pidfd_open is not supported, fallback to kill.  */
       if (errno == ENOSYS)
         return libcrun_kill_linux_no_pidfd (status, true, signal, err);
+      if (errno == ESRCH)
+        return crun_make_error (err, 0, "container not running");
```

Comment:
I think you could also set `errno` right ?

```suggestion
        return crun_make_error (err, errno, "container not running");
```

Reply:
I don't want to add `No such process` after `container not running`, so I kept it `0`